node/src/components/linear_chain_sync.rs

```diff
@@ -11,22 +11,21 @@ pub(crate) use error::LinearChainSyncError;
 pub(crate) use operations::{run_fast_sync_task, KeyBlockInfo};
 
 #[derive(DataSize, Debug)]
+/// Both NotGoingToSync and Done pass onwards the header of the switch block created during those
+/// states.
 pub(crate) enum LinearChainSyncState {
-    NotGoingToSync,
+    NotGoingToSync {
+        switch_block_header: Box<BlockHeader>,
+    },
+    Done {
+        switch_block_header: Box<BlockHeader>,
```

Comment:
I don't think it's always a switch block, i.e. the last block of an era. (Also in `Joiner`.)

```suggestion
        block_header: Box<BlockHeader>,
```

Reply:
Done